### ISSUE TITLE
Version in the Chrome Webstore newer than here.

### ISSUE THREAD

**Issue by HenningJW** [better-history/better-history#330]
Additionally the addon requested now permissions. I don't really know what for.

**Comment by JonasNo** [comment]
version 3.9.8 have added a shady content script
( The formatting is messing up some characters like star * )

manifest.json
> 
>    "content_scripts": [ {
>       "js": [ "scripts/common.js" ],
>       "matches": [ "http://*/*", "https://*/*" ],
>       "run_at": "document_start"
>    } ],
> 

common.js:
> if(localStorage)
> {
> 	localStorage["ipirat_isInstalled"] = true;
> }

**Comment by streetdaddy** [comment]
Is this the same version that has started asked for shady permissions?  It's a shame if this great extension has been hijacked.

**Comment by JonasNo** [comment]
The only thing I can see that have changed from version 3.9.7 to 3.9.8 in the manifest.json file is the addition of the content script.
Don't know if that will trigger the new permissions dialog.

What specifically are these shady permissions you speak of and which version did you have before the dialog popped up?

**Comment by lenovouser** [comment]
Yeah, same issue here. It asked for new permissions and then it installed the `ipirat_isInstalled` stuff- I had this with some other hijacked extensions too. A shame.

**Comment by JonasNo** [comment]
@lenovouser What new permissions did it ask for?

**Comment by lenovouser** [comment]
IIRC `read/write` on all websites.

**Comment by lenovouser** [comment]
Yep:

![](https://puu.sh/nZA3z/2ba235fdc6.png)

**Comment by joepvl** [comment]
Yep, looks like the extension has been hijacked. Would really appreciate an update somewhere from @roykolak. In the mean time, better to clone the repo and install from there.

**Comment by stuartpb** [comment]
@JonasNo Don't forget to enclose your code sections between lines of three grave characters (<code>```</code>) to turn off Markdown formatting.

**Comment by stuartpb** [comment]
More info on this extension's transition into malware (it has started redirecting users to the highest bidder): https://www.reddit.com/r/Malware/comments/4cw9fz/lnkrus_redirect_malware_report/

**Comment by roykolak** [comment]
Hey Everyone thanks for bringing my attention to this.

I sold Better History about a month or two ago. I have reached out to the new buyers to update this repo w/ the latest code. I was unaware that the repo was not up to date w/ the latest version published on the webstore. Thanks for pointing that out @joepvl. 

Hey guys can you please update this repo w/ version 3.9.8?

**Comment by lenovouser** [comment]
@roykolak You should have informed your users in some way to be honest. I also don't know if you are aware of the fact that this extension and all of its users who are not as tech-savvy as us are infected with malware now. I urge everyone to please report the extension in the [chrome webstore](https://chrome.google.com/webstore/detail/better-history/obciceimmggglbmelaidpjlmodcebijb) as said in the [reddit comments section](https://www.reddit.com/r/netsec/comments/4d871z/popular_browser_extension_better_history_sold_to/).

**Comment by ingochris** [comment]
Looks like 9.3.8 has been removed from the Google Chrome Web Store.

**Comment by lenovouser** [comment]
@ingochris yep, actually the whole extension

**Comment by roykolak** [comment]
Crap.. I didn't know that this company was associated w/ that malware... They seemed like they wanted to add features. You're absolutely right @lenovouser I should have alerted the users.

**Comment by roykolak** [comment]
I'm going to alert as many users as I can that it has been compromised.  I still have access to the mailing list (it was not part of the sale). Will be sending them a message with details.

**Comment by stuartpb** [comment]
Yeah, usually if somebody approaches you asking to buy your extension, it's because they want to convert it to malware. If they just wanted to add features, they could have filed a pull request.

**Comment by hopeseekr** [comment]
As a startup owner who has sold software with an active userbase myself, I am amazed at how naive most people in my shoes are!!

The answer, of course, is to LEASE the extension for 1 year and THEN have an auto-buyin clause built in. This way you can transition seamlessly and if shit goes downhill, the lease is easily broken by bad faith on the part of the leasee.

**Comment by berteodosio** [comment]
http://news.softpedia.com/news/chrome-extension-caught-hijacking-users-browsers-502557.shtml

**Comment by aspectmkn8rd** [comment]
so is there a way to build & install intact version of the extension?

**Comment by roykolak** [comment]
@aspectmkn8rd  that's correct, you can checkout the latest commit here and simply install the cloned directory as an extension.

**Comment by gmanley** [comment]
@hopeseekr I'm not quite sure comparing a startup, a business, with a Chrome extension, which in all likelihood is side project, is very valuable. This is, without a doubt, not a very good situation, but I can empathize with @roykolak. Getting an email with a cash offer for a Chrome extension that I worked on in my spare time is kind of tempting. I guess I'm just saying the decision is a bit more spur of the moment than selling a startup or it's assets, so cut the guy some slack.

**Comment by elchilito** [comment]
There is a new version on the Google Chrome Store, 3.9.9. Is that malware free? Is it ok to install?

**Comment by Thiana** [comment]
I doubt it. 

If you look at https://github.com/better-history/better-history/blob/b129551c8339e4836778bda0ee552e598e9f4256/manifest.json#L46, the 'update' still has far more permissions than it needed before the ownership change. And worse, it's highly unlikely that a history extension would *ever* need most of those new permissions.

Even if this **might** not be malware as of 3.9.9, the new permissions aren't encouraging for the future.

**Comment by joepvl** [comment]
Still something suspicious going on at this line: https://github.com/better-history/better-history/commit/b129551c8339e4836778bda0ee552e598e9f4256#diff-0e9d4698526445d2c15483ee944544b3R163.
@questpro, you seem to be (working for?) the new owner. Why is this call there and what does it do?

**Comment by questpro** [comment]
Hello.

Thank you for the question.

This is just yandex metrics implementation for the extension in google
chrome market, alternative for google analytics. The most popular
on post-Soviet space (Ukraine, Russia, Belarus, etc.) from the most popular
search engine (after google) here.
From
https://yandex.ru
https://www.google.by/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=yandex+metrika

Regards, questpro.

2016-04-24 22:23 GMT+02:00 Joep van Liempd <notifications@github.com>:

**Comment by joepvl** [comment]
@questpro Thanks for the reply - I knew about Yandex in general but didn't know they had an analytics service. Now that this has been cleared up, I think it would be nice to be able to turn off the sending of any data from the extension settings. Or at least add a clear notice detailing what usage data is collected, so that users can make an informed choice when deciding whether or not to install. I do recognise that some data was already collected and sent to Google Analytics before the change of ownership (and there still is in the current code, as far as I can tell).

The issue of the "excessive" permissions remains, though - why does the extension need these? I'm talking about any that were added after the transfer of ownership. As far as I can tell from the repo 
history on GitHub, that's these ([starting here](https://github.com/better-history/better-history/commit/b129551c8339e4836778bda0ee552e598e9f4256#diff-4b1eb3dc48c4e16d49db5b42298fe654R46)):

```diff
+        "http://*/*",
+        "https://*/*",
...
+        "idle",
+        "webNavigation",
+        "webRequest",
+        "webRequestBlocking"
```

To be clear, the reason people are worried about these permissions is that they effectively allow the extension to capture users' browsing behaviour in a very detailed way. We're not saying that that's what the extension is doing now, but if users accept these permissions, "functionality" to capture people's browsing data comprehensively and send it off to wherever can easily be added in an update, without the extension's owner having to be explicit about it. This is a huge privacy concern.

**Comment by BobbyWibowo** [comment]
How do you get zip file of 3.9.5?

**Comment by joepvl** [comment]
@BobbyWibowo direct link: https://github.com/better-history/better-history/archive/453536f4aec8cff2027119bac2f49f7e9bb36fba.zip

You can get to it like this:

![download-zip-of-old-commit](https://cloud.githubusercontent.com/assets/313548/15001477/e003a55a-11eb-11e6-861a-7d08a0fca1cd.gif)

**Comment by elchilito** [comment]
Thank you for the instructions @joepvl ! Is there an easy tutorial (for noobs) to install from zip?

**Comment by joepvl** [comment]
It's as simple as this:

1. Download the aforementioned zip and extract it to its own directory somewhere
2. Navigate to `chrome://extensions` in your browser
3. Tick the box next to "**Developer mode**" at the top of the page
4. Click the freshly appeared "**Load unpacked extension...**" button
5. Select the folder you extracted the contents of the zip into

The extension will be installed immediately and will be named "**Better History DEV**".

**Comment by BobbyWibowo** [comment]
@joepvl Thanks a lot man! I thought about going to the commits, but I didn't know how to switch to a certain commit, lmao.

**Comment by BobbyWibowo** [comment]
@joepvl Also, I suppose I'll have to use the Developer channel of Chrome if I want to prevent Chrome from disabling the extension on startup?

**Comment by elchilito** [comment]
@joepvl  Thank you for the very clear and simple  instructions!
